rviz_visual_testing_framework/src/visual_test_fixture.cpp

```diff
@@ -53,11 +53,11 @@ void VisualTestFixture::SetUpTestCase()
   visualizer_app_->init(argc, argv);
   if (VisualTest::generateReferenceImages()) {
     visualizer_app_->loadConfig(QDir::toNativeSeparators(
-        QString::fromStdString(std::string(_SRC_DIR_PATH) + "/visual_tests_default_config.rviz")));
+        QString::fromStdString(std::string(_SRC_DIR_PATH) + "/config/visual_tests_default_config.rviz")));
```

Comment:
Here (as well as in the `else` case, lines 59-60) the installed configs should be used, not the ones in the source directory.
This means that one has do define in the CMakeLists.txt a Path variable pointing to the install folder, similarly to what done with `_SRC_DIR_PATH`, and use that instead of `_SRC_DIR_PATH` here, otherwise the problem exposed in #372 is not solved, at least according to my understanding. Or have you successfully run the visual tests after having built the `rviz_visual_testing_framework` with these changes and having removed the source directory?

Reply:
Thanks for pointing this out. I will clean my workspace and try again. On this note, would it be better to use ```ros::package:getPath``` and remove ```_SRC_DIR_PATH``` from CMakeLists?

Reply:
The point is that if I, for example, install `rviz_visual_testing_framework` form binaries (e.g. via the Debian package) I don't have the source repository, so that `_SRC_DIR_PATH` as is now defined in the CMakeLists, will be a non-existing path. That's why it must be replaced by a path to the installed configs.

Concerning the `ros::package::Path`, I guess it doesn't exist in ROS2, the corresponding functionality is provided by the resource index (see [here](https://answers.ros.org/question/288501/ros2-equivalent-of-rospackagegetpath/)).

I think this is a good alternative to defining the path variable in the CMakeLists.txt. Which one of the two approaches is better, though, I think could probably be judged better by somebody like @wjwwood.

Reply:
Yeah that's fine, but if you're using them in the package which installs them, then you may run into issues trying to run the tests. You may need special handling to use the file within the package which installs them. I can't say off-hand what exactly will happen, but in my experience you'll run into issues, but you can work around them.

For other packages which want to use the configs, the proposed approach will work well.

Reply:
I need to look more into this, but I think the lines above are just frame names when rviz is launched. I can run tests from rviz_default_plugins even if the config files are not available in source or install folder, so this PR may be pointless.

Reply:
> Yeah that's fine, but if you're using them in the package which installs them, then you may run into issues trying to run the tests. 

I also thought about that, but it shouldn't be a problem, because there are no visual tests in the package `rviz_visual_testing_framework`, and there is no reason for some to be added in the future, given that this is not a package supposed to contain displays. Currently all the default visual tests are in `rviz_default_plugins` and if someone writes their own display and wants to have visual tests for it, they would write them in their own package.

Reply:
> I need to look more into this, but I think the lines above are just frame names when rviz is launched. I can run tests from rviz_default_plugins even if the config files are not available in source or install folder, so this PR may be pointless.

This is not true: the property of the tests configs are fundamental for the visual tests to work (e.g. there must be no grid display pre loaded and the windows dimensions also play an important role). If the configs are not available, then RViz will use the default config instead (or the last one saved by the user) and this will result in the failure of the tests.

Reply:
Thank you. This makes sense that it was using the default rviz.